client/my-sites/domains/domain-management/components/primary-domain-selector/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { FEATURE_SET_PRIMARY_CUSTOM_DOMAIN } from '@automattic/calypso-products';
+import { useTranslate } from 'i18n-calypso';
+import { useState, ChangeEvent, useEffect } from 'react';
+import FormSelect from 'calypso/components/forms/form-select';
+import type { DomainData, SiteDetails } from '@automattic/data-stores';
+
+import './style.scss';
+
+type PrimaryDomainSelectorProps = {
+	domains: undefined | DomainData[];
+	userCanSetPrimaryDomains: boolean;
+	site: undefined | null | SiteDetails;
+	onSetPrimaryDomain: ( domain: string, onComplete: () => void ) => void;
+};
+
+const PrimaryDomainSelector = ( {
+	domains,
+	site,
+	userCanSetPrimaryDomains,
+	onSetPrimaryDomain,
+}: PrimaryDomainSelectorProps ) => {
+	const [ selectedDomain, setSelectedDomain ] = useState< undefined | string >( undefined );
+	const [ isSettingPrimaryDomain, setIsSettingPrimaryDomain ] = useState< boolean >( false );
+
+	const translate = useTranslate();
+
+	useEffect( () => {
+		if ( domains?.length ) {
+			const primaryDomain = domains.find( ( domain ) => {
+				return domain.primary_domain;
+			} );
+			if ( primaryDomain ) {
+				setSelectedDomain( primaryDomain.domain );
+			}
+		}
+	}, [ domains ] );
+
+	if ( ! domains || ! site ) {
+		return null;
+	}
+
+	const isOnFreePlan = site?.plan?.is_free ?? false;
+
+	const shouldUpgradeToMakeDomainPrimary = ( domain: DomainData ): boolean => {
+		return (
+			! userCanSetPrimaryDomains &&
+			isOnFreePlan &&
```

Comment:
I'm a bit confused on the logic here. If you're checking `isOnFreePlan`, won't that override PWPO users who are entitled to set a primary domain while still being on a Free plan?

Reply:
For PWPO users `userCanSetPrimaryDomains` will be `true`, so `shouldUpgradeToMakeDomainPrimary` will return `false`. It's the same logic we have in the table to enable the set primary address action.

Reply:
Ahhh, got it! Thanks for explaining! :)